src/lib.rs

```diff
@@ -215,15 +215,39 @@ impl GeLineq {
         None
         
     }
-    
+
     pub fn substitution(main_gelineq: &GeLineq, variable_index: u32, sub_gelineq: &GeLineq) -> Option<GeLineq> {
+        let mut var_to_substitute: usize = main_gelineq.indices.len();
+        for i in 0..main_gelineq.indices.len(){
```

Comment:
`let var_to_substitute = main_gelineq.indices.iter().position(|x| x == variable_index).expect(&format!("Variable '{variable_index}' to substitute not found"))`

instead of lines 220-229